.github/workflows/update_golang_dependencies.yml

```diff
@@ -0,0 +1,59 @@
+name: Update Golang Dependencies
+
+on:
+  schedule:
+    - cron: "0 0 * * *" # Runs every day at midnight UTC
+  workflow_dispatch:
+
+permissions: read-all
+
+jobs:
+  update_golang_deps:
+    if: github.repository == 'vitessio/vitess'
+    permissions:
+      contents: write
+      pull-requests: write
+    strategy:
+      matrix:
+        branch: [ main, release-18.0, release-17.0, release-16.0 ]
+    name: Update Golang Dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21.5
```

Comment:
How will we keep this aligned with the go version as we upgrade that pretty regularly, especially on main.

If we do end up doing this on release branches, will this not cause issues when those are using an older version of go than main?

Reply:
The automation that bumps the golang version throughout the codebase will keep this aligned for us without having to do anything more.